clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java

```diff
@@ -302,10 +444,16 @@ public ServerStoreImpl createStore(String name, ServerStoreConfiguration serverS
     PageSource resourcePageSource = getPageSource(name, serverStoreConfiguration.getPoolAllocation());
     ServerStoreImpl serverStore = new ServerStoreImpl(serverStoreConfiguration, resourcePageSource);
     stores.put(name, serverStore);
+
+    registerStoreStatistics(name);
+
     return serverStore;
   }
 
   public void destroyServerStore(String name) throws ClusterException {
+
+    unRegisterStoreStatistics(name);
```

Comment:
so unRegisterStoreStatistics(name); is called indirectly via EhcacheActiveEntity.destroy.  This is the call chain:
1. EhcacheActiveEntity.destroy()
2. ehcacheStateService.destroy()
3. releasePools(...)
4. releasePool(...)
5. unRegisterPoolStatistics(...);
Let me know if you want it done differently?